drivers/watchdog/wdt_nxp_ewm.c

```diff
@@ -163,9 +163,11 @@ static int nxp_ewm_init(const struct device *dev)
 	}
 #endif /* DT_INST_NODE_HAS_PROP(0, clk_sel) */
 
+#if DT_INST_NODE_HAS_PROP(0, clk_divider)
 	if (config->clk_divider >= 0 && config->clk_divider <= 0xFF) {
 		base->CLKPRESCALER = EWM_CLKPRESCALER_CLK_DIV(config->clk_divider);
 	}
+#endif /* DT_INST_NODE_HAS_PROP(0, clk_divider) */
```

Comment:
actually, I suggest not to introduce this #ifdef , especially since it is only single instance but the driver was already written multi-instance, instead you can use DT_INST_PROP_OR in the init macro to make it 0 if not present on DT and this code already checks on that. This is actually necessary anyways since you made it `required: false` now.

Reply:
clk_divider's type is `uint8_t` and the code accept 0-255 here, so no usable fallback for `DT_INST_PROP_OR`.

Reply:
@decsny what do you think if I make the driver explicitly single instance? Not only for the init macro discussed in this thread, but also for adopting clock control APIs for supporting platforms.